packages/Operators/test/reference_implementation/DTK_ReferenceHexImpl.cpp

```diff
@@ -0,0 +1,184 @@
+// Constructor.
+ReferenceHexImpl::ReferenceHexImpl(
+    const int id,
+    const int owner_rank,
+    const Teuchos::Array<DataTransferKit::Entity>& nodes )
+    : d_extra_data( new ReferenceHexExtraData() )
+{
+    // Get the element id and owner rank.
+    d_extra_data->id = id;
+    d_extra_data->owner_rank = owner_rank;
+
+    // Get the element node ids.
+    d_extra_data->node_ids.resize( 8 );
+    for ( int n = 0; n < 8; ++n )
+    {
+        d_extra_data->node_ids[n] = nodes[n].id();
+    }
+
+    // Get the element node coordinates.
+    Teuchos::Tuple<double,6> node_bounds;
+    DTK_REQUIRE( 8 == nodes.size() );
+    auto& coords = d_extra_data->node_coords;
+    coords.resize( 1, 8, 3 );
+    for ( int n = 0; n < 8; ++n )
+    {
+        nodes[n].boundingBox( node_bounds );
+        for ( int d = 0; d < 3; ++n )
+        {
+            coords(0,n,d) = node_bounds[d];
+        }
+    }
+}
+
+//---------------------------------------------------------------------------//
+// Get the unique global identifier for the entity.
+DataTransferKit::EntityId ReferenceHexImpl::id() const
+{ 
+    return d_extra_data->id;
+}
+    
+//---------------------------------------------------------------------------//
+// Get the parallel rank that owns the entity.
+int ReferenceHexImpl::ownerRank() const
+{ 
+    return d_extra_data->owner_rank;
+}
+
+//---------------------------------------------------------------------------//
+// Get the topological dimension of the entity.
+int ReferenceHexImpl::topologicalDimension() const
+{
+    return 3;
+}
+
+//---------------------------------------------------------------------------//
+// Return the physical dimension of the entity.
+int ReferenceHexImpl::physicalDimension() const
+{ 
+    return 3;
+}
+
+//---------------------------------------------------------------------------//
+// Return the Cartesian bounding box around an entity.
+void ReferenceHexImpl::boundingBox( Teuchos::Tuple<double,6>& bounds ) const
+{
+    double max = std::numeric_limits<double>::max();
+    bounds = Teuchos::tuple( max, max, max, -max, -max, -max );
+    for ( int n = 0; n < 8; ++n )
+    {
+	for ( int d = 0; d < 3; ++d )
+	{
+	    bounds[d] = std::min( bounds[d],
+                                  d_extra_data->node_coords(0,n,d) );
+	    bounds[d+3] = std::max( bounds[d+3],
+                                    d_extra_data->node_coords(0,n,d) );
+	}
+    }
+}
+
+//---------------------------------------------------------------------------//
+// Determine if an entity is in the block with the given id.
+bool ReferenceHexImpl::inBlock( const int block_id ) const
+{
+    // no blocks for now.
+    return false;
+}
+
+//---------------------------------------------------------------------------//
+// Determine if an entity is on the boundary with the given id.
+bool ReferenceHexImpl::onBoundary( const int boundary_id ) const
+{
+    // no boundaries for now.
+    return false;
+}
```

Comment:
This mesh currently doesn't support blocks and boundaries so I guess throwing would also work. Technically because nothing is in a a block or on a boundary returning false should be valid.